### DISCUSSION TITLE
playing montage with set locomotion action notify twice in a row

### DISCUSSION THREAD

**Discussion by nekwo** [Sixze/ALS-Refactored#250] [Q&A] [answered]
playing/blending two different montages with set locomotion action notify sets the locomotion action back to default pose

**Comment by nekwo** [comment]
![image](https://user-images.githubusercontent.com/43717185/193118088-4ddfccf9-1698-40ef-829f-ee25b3a307f8.png)

this is one method to get around it on the second montage but I want to see if there is a code fix since this may not be okay for many situations

**Comment by Sixze** [comment, marked as answer]
This is a common issue with animation notifies. During blending, there is no guarantee that the animation notify in the first animation will end before the similar notify in the second animation starts, so a conflict occurs.

Such issues are best solved not at the level of animations, but at the level of logic that controls the playback of these animations. For example, you can use the Gameplay Ability System and change locomotion actions from abilities that play these animations instead of using `UAlsAnimNotifyState_SetLocomotionAction`.

In any case, in the future I plan to get rid of locomotion actions and replace them with the Gameplay Ability System, so it is very likely that `UAlsAnimNotifyState_SetLocomotionAction` will simply disappear.
